server/webapp/WEB-INF/rails.new/app/controllers/api_v3/admin/environments_controller.rb

```diff
@@ -22,7 +22,19 @@ class EnvironmentsController < BaseController
       before_action :check_for_stale_request, only: [:put]
 
       def index
-        render DEFAULT_FORMAT => Admin::Environments::EnvironmentsConfigRepresenter.new(environment_config_service.getEnvironments()).to_hash(url_builder: self)
+        is_query_param_provided = params.length > 2
+        is_with_remote = params[:withconfigrepo]
+        environment_names = environment_config_service.environmentNames()
+
+        if is_with_remote.nil? and !is_query_param_provided
+          load_local_environments(environment_names)
+        elsif is_with_remote and is_with_remote.downcase == 'true'
+          load_merged_environments(environment_names)
+        else
+          return render_not_found_error
+        end
```

Comment:
You may be able to simplify a lot of this if you simply ask the environment config service. Additionally, since all the tests are simply mocks, it's easier to read the tests too..

```ruby
environment_config_service.getLocalEnvironments()
environment_config_service.getMergedEnvironments()
```